db/change_cache.go

```diff
@@ -131,12 +132,14 @@ func (c *changeCache) Init(context *DatabaseContext, notifyChange func(base.Set)
 
 	heap.Init(&c.pendingLogs)
 
-	// Start a background task for periodic housekeeping:
+	// Start a background task for skipped sequence pending logs
+	// and pruning the cache based on length
 	go func() {
 		for {
 			select {
 			case <-time.After(c.options.CachePendingSeqMaxWait / 2):
-				c.CleanUp()
+				c.InsertPendingEntries()
+				c.CleanUpLength()
```

Comment:
Can remove CleanUpLength - will be automatically triggered on cache add in InsertPendingEntries.